Regex changes to be less strict with syntax

davisben85 opened the pull request:
The previous regex required that the SQL generated include `'s around ids, which makes the parse fragile.  I changed the parse to look for any non-whitespace string after the ALTER TABLE.  This will support ALTER TABLE `foo` and ALTER TABLE foo , now.

I have also added an additional sub to the parsed_statement, for users utilizing strip_heredoc (has \n in the SQL) instead of squish (flattens the SQL), again, so this parser is less fragile.

sauloperez commented:
Hi @davisben85 and thanks for your contribution! I would add some tests to ensure what you claim.

davisben85 commented:
@sauloperez are you referring to adding some stuff to the test_database.rb to exercise the changes?

davisben85 commented:
Never mind, after a quick browse, I see where I can add some checks in to verify.

davisben85 commented:
I've added tests for the changes.  All current tests that ran were passing.  I also changed the regex again, to better handle certain situations with the line feed being adjacent to the table name.

davisben85 commented:
@sauloperez alrighty... things are passing.  I had to update the travis config, since they went to 16.04 following the discontinued support for 14.04 back in April.  The tests I added to verify the regex changes passed as well, so it should support IDs with and without grave marks.

davisben85 commented:
@sauloperez let me know if there's anything that you'd like me to include in this PR to get it merged.
